cmd/spire-server/cli/run/run_posix_test.go

```diff
@@ -267,3 +275,18 @@ func newServerConfigCasesOS() []newServerConfigCase {
 func testParseConfigGoodOS(t *testing.T, c *Config) {
 	assert.Equal(t, c.Server.SocketPath, "/tmp/spire-server/private/api-test.sock")
 }
+
+func getAvailablePort() (string, error) {
```

Comment:
Ugh - a classic port collision. As written, this `getAvailablePort()` could potentially still race against other processes. The kernel could reallocate the port between the time the deferred `l.Close()` occurs and the time that the test code attempts to bind to the port returned from this helper.

Also nit: should mark this as a helper with `t.Helper()`.